### ISSUE TITLE
Context does not persist when switching projects

### ISSUE THREAD

**Issue by Jundor-y** [node-red/node-red#3809]
### Current Behavior

When switching between different projects, Nodered keeps deleting the projects content, so anything set with content.set("ID", 123, "file") gets deleted.

### Expected Behavior

I'd expect the content to be persistent, even when switching between projects.

### Steps To Reproduce

Create a project
Use a node to set the content, i.e. a function node with context.set("ID", 123, "file");
// Under /context a new folder is created
**Either**:
Create a second project
**Or**:
Open another project and deselect "Clear context when switching between projects"

// The previous context is now deleted

### Example flow

```
[{"id":"cc7ba6850f1678f0","type":"tab","label":"Flow 1","disabled":false,"info":"","env":[]},{"id":"bf973ee7e6b9682f","type":"inject","z":"cc7ba6850f1678f0","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":true,"onceDelay":"1","topic":"","payload":"","payloadType":"date","x":110,"y":120,"wires":[["f48b6e7ceefc1b5e"]]},{"id":"f48b6e7ceefc1b5e","type":"function","z":"cc7ba6850f1678f0","name":"function 1","func":"return msg;","outputs":1,"noerr":0,"initialize":"// Code added here will be run once\n// whenever the node is started.\n context.set(\"ID\", 123, \"file\");","finalize":"","libs":[],"x":280,"y":120,"wires":[[]]}]```


### Environment

- Node-RED version: v3.0.1
- Node.js version: v16.16.0
- npm version:
- Platform/OS: Linux 4.15.0 x64 LE
- Browser: Independent of browser, the issue occurs on all Safari, Chrome and Firefox.

**Comment by DmitrySidorow** [comment]
agreed. I think the context file should be separate for each project.

**Comment by juggledad** [comment]
This also effects flow context and has been happening since at least v2.0.0 (I went back and tried it there and in 2.2.2)
note in v2.0.0 there was no checkbox, but the checkbox does exist in v2.2.2 and the issue exist too.

**Comment by knolleary** [comment]
@juggledad it always used to be the behaviour of NR that we would delete context when switch projects. That was the well-defined and expected behaviour. So to say the behaviour is the same in versions that didn't give the option *not* to delete context is not actually adding any new information here.

The point is we added the option to not delete context in a more recent 2.x release (I forget when exactly) - and the issue here is a claim that option isn't being honoured properly.

**Comment by knolleary** [comment]
@DmitrySidorow changing context to be project-aware is a different matter entirely and would require significant, breaking changes to how context works today. That isn't in scope of this bug fix.

**Comment by knolleary** [comment]
@Jundor-y  Looking at your example flow, you are setting a context value on a specific node. Do you have a node with the same ID in the project you are switching to?

Node-RED will always clean up context for nodes that no longer exist in the deployed flows. The "Clear context when switching between projects" option selects whether global context gets cleared, as well as the context for an entities that also exist in the new project.

Creating a new Project, for example, will mean there are no nodes/flows with matching IDs, so the context *will* get cleared for them. That is the expected correct behaviour.

**Comment by knolleary** [comment]
Closing as part of a tidy-up of the issue tracker — this has been inactive for a long time and much has changed in Node-RED since. If it's still relevant on a current version, please open a fresh issue with up-to-date details and we'll take a look. Thanks for raising it.
